app/lib/report_helper.rb

```diff
@@ -427,10 +427,26 @@ def top_taxons_details(samples, background_id, num_results, sort_by_key, only_sp
       taxon_counts = res["taxon_counts"]
       sample_id = pr.sample_id
       tax_2d = convert_2d(taxon_counts)
+      if only_species
+        puts "ONLY SPECIES"
+        puts "TAX LEVEL SPECIES: " + TaxonCount::TAX_LEVEL_SPECIES.to_s
+        tax_2d.keep_if { |_tax_id, tax_info| tax_info['tax_level'] = TaxonCount::TAX_LEVEL_SPECIES }
```

Comment:
@jshoe don't you mean '==' instead of '=' here?

Reply:
also this filter needs to be applied after cleaning and name validation, not before

Reply:
and you need to get rid of the filter in the sql query above

Reply:
holy the "==" was my main bug lol

Reply:
:)